R/convert_mae_se_assay_to_dt.R

```diff
@@ -39,8 +39,12 @@ convert_se_assay_to_dt <- function(se,
   checkmate::assert_flag(retain_nested_rownames)
   validate_se_assay_name(se, assay_name)
   if (wide_structure) {
-    # wide_structure works only with `normalization_type` column in the assay
-    if ("normalization_type" %in%
+    # wide_structure works only with `normalization_type` column in the assay 
+    # and only for assays class "BumpyMatrix"
+    if (!inherits(SummarizedExperiment::assay(se, assay_name), "BumpyDataFrameMatrix")) {
+      warning("assay is not class `BumpyMatrix`, wide_structure=TRUE ignored")
```

Comment:
What assays do not follow `BumpyMatrix` structure?

Reply:
[In our test](https://github.com/gdrplatform/gDRutils/blob/master/tests/testthat/test-convert_mae_se_assay_to_dt.R#L87-L164) 😉